apps/mobile/src/screens/NftDetailScreen/NftAdditionalDetailsEth.tsx

```diff
@@ -0,0 +1,119 @@
+import { useMemo } from 'react';
+import { View } from 'react-native';
+import { useFragment } from 'react-relay';
+import { graphql } from 'relay-runtime';
+
+import { NftAdditionalDetailsEthFragment$key } from '~/generated/NftAdditionalDetailsEthFragment.graphql';
+import { getOpenseaExternalUrl, hexHandler } from '~/shared/utils/getOpenseaExternalUrl';
+
+import { EnsOrAddress } from '../../components/EnsOrAddress';
+import { InteractiveLink } from '../../components/InteractiveLink';
+import { LinkableAddress } from '../../components/LinkableAddress';
+import { DetailLabelText, DetailSection, DetailValue } from './DetailSection';
+
+type NftAdditionalDetailsEthProps = {
+  showDetails: boolean;
+  tokenRef: NftAdditionalDetailsEthFragment$key;
+};
+
+export function NftAdditionalDetailsEth({ tokenRef, showDetails }: NftAdditionalDetailsEthProps) {
+  const token = useFragment(
+    graphql`
+      fragment NftAdditionalDetailsEthFragment on Token {
+        externalUrl
+        tokenId
+
+        chain
+
+        owner {
+          username
+        }
+
+        contract {
+          creatorAddress {
+            address
+            ...LinkableAddressFragment
+            ...EnsOrAddressWithSuspenseFragment
+          }
+          contractAddress {
+            address
+            ...LinkableAddressFragment
+            ...EnsOrAddressWithSuspenseFragment
+          }
+        }
+      }
+    `,
+    tokenRef
+  );
+
+  const { tokenId, contract, externalUrl } = token;
+
+  const openSeaExternalUrl = useMemo(() => {
+    if (contract?.contractAddress?.address && tokenId) {
+      return getOpenseaExternalUrl(contract.contractAddress.address, tokenId);
+    }
+
+    return null;
+  }, [contract?.contractAddress?.address, tokenId]);
+
+  return (
+    <View className="flex flex-col space-y-4">
+      {token.owner?.username && (
+        <DetailSection>
+          <DetailLabelText>OWNED BY</DetailLabelText>
+
+          {/* TODO(Terence) When the user profile screen is ready, setup the onPress here */}
```

Comment:
you tracking this somewhere?

Reply:
https://linear.app/galleryso/issue/GAL-2782/wire-up-links-from-other-screens